lib/browser/evaluate.js

```diff
@@ -56,3 +56,26 @@ exports.evaluate = function evaluate() {
       throw new Error(invocation);
   }
 };
+
+exports.evaluateAsync = function evaluateAsync() {
+  const args = _.toArray(arguments);
+  let clientFunction = args.pop();
+
+  const invocation =
+    'evaluateAsync(clientFunction) - requires (Function|String) clientFunction';
+  assert.truthy(invocation, clientFunction);
+
+  switch (typeof clientFunction) {
+    case 'function':
+      clientFunction = `var done = arguments[arguments.length-1], args=[].slice.call(arguments, 0, arguments.length-1);
+      return new Promise(
+        (resolve) => resolve((${clientFunction}).apply(this, args))
+      ).then(done)`;
```

Comment:
Do we need the `new Promise()` here?  Maybe if we want to make sure it's a Promise, just:

```js
        return Promise.resolve((${clientFunction}).apply(this, args)).then(done)
```

Reply:
It depends on the behavior we want for uncaught/sync errors. Errors thrown within the promise constructor are turned into rejections.